core/src/test/java/apoc/trigger/TriggerNewProceduresTest.java

```diff
@@ -0,0 +1,583 @@
+                                    return relIterator.hasNext();
+//                                            && relIterator.next().getAllProperties().equals(expectedProps);
+                                })
+                , (value) -> value, 30L, TimeUnit.SECONDS);
+    }
+
+    @Test
+    public void testDeleteRelationships() {
+        db.executeTransactionally("CREATE (a:A {name: \"A\"})-[:R1]->(z:Z {name: \"Z\"}), (a)-[:R2]->(z)");
+        String name = "trigger-after-async-3";
+        final String query = "UNWIND $deletedRelationships AS r\n" +
+                "MERGE (a:AA{name: \"AA\"})\n" +
+                "SET a.triggerAfter = size($deletedRelationships) = 1, a.deleted = type(r)";
+        sysDb.executeTransactionally("CALL apoc.trigger.install('neo4j', $name, $query, {phase: 'after'})",
+                Map.of("name", name, "query", query));
+        awaitTriggerDiscovered(db, name, query);
+
+
+        db.executeTransactionally("MATCH (a:A {name: \"A\"})-[r:R2]->(z:Z {name: \"Z\"})\n" +
+                "DELETE r");
+
+        assertEventually(() ->
+                        db.executeTransactionally("MATCH (a:AA) RETURN a", Map.of(),
+                                result -> {
+                                    final Node r = result.<Node>columnAs("a").next();
+                                    return (boolean) r.getProperty("triggerAfter", false)
+                                            && r.getProperty("deleted", "").equals("R2");
+                                })
+                , (value) -> value, 30L, TimeUnit.SECONDS);
+    }
+    
+    //
+    // new test cases
+    //
+
+    @Test
+    public void testTriggerShow() {
+        String name = "test-show1";
+        String name2 = "test-show2";
+        String query = "MATCH (c:TestShow) SET c.count = 1";
+        
+        testCall(sysDb, "CALL apoc.trigger.install('neo4j', $name, $query,{}) YIELD name",
+                map("query", query, "name", name),
+                r -> assertEquals(name, r.get("name")));
+        
+        testCall(sysDb, "CALL apoc.trigger.install('neo4j', $name, $query,{}) YIELD name",
+                map("query", query, "name", name2),
+                r -> assertEquals(name2, r.get("name")));
+        
+        // not updated
+        testResult(sysDb, "CALL apoc.trigger.show('neo4j') " +
+                        "YIELD name, query RETURN * ORDER BY name",
+                map("query", query, "name", name),
+                res -> {
+                    Map<String, Object> row = res.next();
+                    assertEquals(name, row.get("name"));
+                    assertEquals(query, row.get("query"));
+                    row = res.next();
+                    assertEquals(name2, row.get("name"));
+                    assertEquals(query, row.get("query"));
+                    assertFalse(res.hasNext());
+                });
+    }
+    
```

Comment:
Could we add a test to make sure apoc.trigger.show only lists the triggers from the right database?

- create trigger for db "neo4j"
- create trigger for db "foo"
- show("neo4j") only shows neo4j
- show("foo") only shows foo

Reply:
If you are happy to add this test, would you be happy to then back-port it to 4.4?

Reply:
The `TestDatabaseManagementServiceBuilder` is a community version so it doesn't support the "CREATE DATABASE dbName" command, nor the `DatabaseManagementService.createDatabase(..)` method.

Afaik I don't think it's easy (or feasible) to do it here. 
It would be better to create a new test class with an enterprise docker instance or reuse `TriggerClusterRoutingTest`.

And yes, we could back-port it to 4.4 as well.

Reply:
There are 3 comments where we basically touch on this idea of "we should use the clustering test file, or we should make a new test file with an enterprise container".

I think for all 3 comments it would be nicer to not use the cluster, because it's incredibly slow to spin up and is unfortunately quite flaky. I think instead we should either modify the existing TriggerNewProceduresTest.java file to be dockerized neo4j rather than use the in-memory neo4j , or that we should make a new test file for enterprise tests which uses a dockerized neo4j.

Reply:
I added a new test `TriggerEnterpriseFeaturesTest`. 
Personally I preferred to leave the test as before for consistency with the `TriggerTest` (so we are sure that everything works correctly with neo4j community too).
But mostly because it's lighter. 

Also other tests, like `apoc.schema.`, have 2 test classes, 
one with community and the other with enterprise (only with features not available in community)